crates/source_analyzer/src/manifest.rs

```diff
@@ -0,0 +1,133 @@
+use std::{
+    fmt,
+    fs::read_to_string,
+    path::{Path, PathBuf},
+};
+
+use serde::{
+    de::{value::MapAccessDeserializer, MapAccess, Visitor},
+    Deserialize, Deserializer,
+};
+
+use crate::{cycler::CyclerKind, error::Error};
+
+#[derive(Deserialize, Debug)]
+pub struct FrameworkManifest {
+    pub cyclers: Vec<CyclerManifest>,
+}
+
+impl FrameworkManifest {
+    pub fn try_from_toml(path: impl AsRef<Path>) -> Result<Self, Error> {
+        let path = path.as_ref();
+        toml::from_str(&read_to_string(path).map_err(|error| Error::Io {
+            source: error,
+            path: path.to_path_buf(),
+        })?)
+        .map_err(|error| Error::ConfigurationParsing {
+            source: error,
+            path: path.to_path_buf(),
+        })
```

Comment:
All parsing and error generation has been removed from this file.

Reply:
Yes, but please apply this suggestion to `crates/source_analyzer/nodes.rs`, s.t. all originating errors are named `source` because this will be used by `thiserror` as context information.

Reply:
Ah, didn't know that was a magic name. Renamed.